js/plugins/google-genai/src/googleai/gemini.ts

```diff
@@ -0,0 +1,794 @@
+
+export const GeminiConfigSchema = GenerationCommonConfigSchema.extend({
+  apiKey: z
+    .string()
+    .describe('Overrides the plugin-configured API key, if specified.')
+    .optional(),
+  safetySettings: z
+    .array(SafetySettingsSchema)
+    .describe(
+      'Adjust how likely you are to see responses that could be harmful. ' +
+        'Content is blocked based on the probability that it is harmful.'
+    )
+    .optional(),
+  codeExecution: z
+    .union([z.boolean(), z.object({}).strict()])
+    .describe('Enables the model to generate and run code.')
+    .optional(),
+  contextCache: z
+    .boolean()
+    .describe(
+      'Context caching allows you to save and reuse precomputed input ' +
+        'tokens that you wish to use repeatedly.'
+    )
+    .optional(),
+  functionCallingConfig: z
+    .object({
+      mode: z.enum(['MODE_UNSPECIFIED', 'AUTO', 'ANY', 'NONE']).optional(),
+      allowedFunctionNames: z.array(z.string()).optional(),
+    })
+    .describe(
+      'Controls how the model uses the provided tools (function declarations). ' +
+        'With AUTO (Default) mode, the model decides whether to generate a ' +
+        'natural language response or suggest a function call based on the ' +
+        'prompt and context. With ANY, the model is constrained to always ' +
+        'predict a function call and guarantee function schema adherence. ' +
+        'With NONE, the model is prohibited from making function calls.'
+    )
+    .optional(),
+  responseModalities: z
+    .array(z.enum(['TEXT', 'IMAGE', 'AUDIO']))
+    .describe(
+      'The modalities to be used in response. Only supported for ' +
+        "'gemini-2.0-flash-exp' model at present."
+    )
+    .optional(),
+  googleSearchRetrieval: z
+    .union([z.boolean(), z.object({}).passthrough()])
+    .describe(
+      'Retrieve public web data for grounding, powered by Google Search.'
+    )
+    .optional(),
+}).passthrough();
+export type GeminiConfig = z.infer<typeof GeminiConfigSchema>;
+
+export const GeminiTtsConfigSchema = GeminiConfigSchema.extend({
+  speechConfig: z
+    .object({
+      voiceConfig: VoiceConfigSchema.optional(),
+      multiSpeakerVoiceConfig: z
+        .object({
+          speakerVoiceConfigs: z
+            .array(
+              z
+                .object({
+                  speaker: z.string().describe('Name of the speaker to use'),
+                  voiceConfig: VoiceConfigSchema,
+                })
+                .describe(
+                  'Configuration for a single speaker in a multi speaker setup'
+                )
+                .passthrough()
+            )
+            .describe('Configuration for all the enabled speaker voices'),
+        })
+        .describe('Configuration for multi-speaker setup')
+        .passthrough()
+        .optional(),
+    })
+    .describe('Speech generation config')
+    .passthrough()
+    .optional(),
+}).passthrough();
+
+const gemini15Pro = modelRef({
```

Comment:
could you please dedupe these.
 - drop 1.5 models
 - drop `versions` from metadata
 - don't export anything.

Reply:
done